prometheus_api_client/metric.py

```diff
@@ -68,9 +69,13 @@ def __init__(self, metric, oldest_data_datetime=None):
             )
             self.metric_values["ds"] = pandas.to_datetime(self.metric_values["ds"], unit="s")
 
-        # Set the metric start time and the metric end time
-        self.start_time = dateparser.parse(str(self.metric_values["ds"].head(1).values[0]))
-        self.end_time = dateparser.parse(str(self.metric_values["ds"].tail(1).values[0]))
+        # Set the metric start time and the metric end time from timestamps in ns
+        self.start_time = datetime.fromtimestamp(
+            self.metric_values["ds"].head(1).values[0].astype(int) * (10 ** -9)
```

Comment:
Are you trying to get the first element of the column "ds" in the dataframe self.metric_values? 

If so, it can be done without calling `.head()`, `self.metric_values["ds"].values[0].astype(int)` should work just as well.

Reply:
I thought about it, but if I do that it will convert the whole df into a numpy array and I will just use 1 or 2 values from the array. The way I did it, it selects only the first value and converts it into an array

Reply:
Another way that avoids generating an entire numpy array would be, `self.metric_values.iloc[0,0].astype(int)` since we know "ds" is always the zeroth column.

Reply:
I am directly using the numpy.datetime64 object instead of converting it to timestamp and then to datetime.datetime